__mocks__/react-native.js

```diff
@@ -0,0 +1,64 @@
+// eslint-disable-next-line no-restricted-imports
+import * as ReactNative from 'react-native';
+import _ from 'underscore';
+
+jest.doMock('react-native', () => {
+    let url = 'https://new.expensify.com/';
+    const getInitialURL = () => Promise.resolve(url);
+
+    let appState = 'active';
+    let count = 0;
+    const changeListeners = {};
+    return Object.setPrototypeOf(
+        {
+            NativeModules: {
+                ...ReactNative.NativeModules,
+                BootSplash: {
+                    getVisibilityStatus: jest.fn(),
+                    hide: jest.fn(),
+                },
+                StartupTimer: {stop: jest.fn()},
+            },
+            Linking: {
+                ...ReactNative.Linking,
+                getInitialURL,
+                setInitialURL(newUrl) {
+                    url = newUrl;
+                },
+            },
+            AppState: {
+                ...ReactNative.AppState,
+                get currentState() {
+                    return appState;
+                },
+                emitCurrentTestState(state) {
+                    appState = state;
+                    _.each(changeListeners, listener => listener(appState));
+                },
+                addEventListener(type, listener) {
+                    if (type === 'change') {
+                        const originalCount = count;
+                        changeListeners[originalCount] = listener;
+                        ++count;
+                        return {
+                            remove: () => {
+                                delete changeListeners[originalCount];
+                            },
+                        };
+                    }
+
+                    return ReactNative.AppState.addEventListener(type, listener);
+                },
+            },
+            Dimensions: {
+                ...ReactNative.Dimensions,
+                addEventListener: jest.fn(),
+                get: () => ({
+                    // Tests will run with the app in a typical small screen size
```

Comment:
Is there a reason or purpose for being a small screen? Is this something that maybe a test would be able to specify or not (since it can change the behavior of components)?

Reply:
Great question. I'm not sure yet. I chose to hardcode in the "small screen" since I wanted to primary make sure we are covering the native behavior and because the react-native test renderer is always pulling from `index.native.js` source.

I was thinking a good future improvement would be to allow some way to do something like:

```js
Dimensions.set(TestHelper.SCREEN_SIZE.SMALL);
```

Reply:
I went ahead and did this. Even though no one is using it yet we'll have the option in the future to test on a non small screen.